lib/buffer.js

```diff
@@ -97,13 +114,22 @@ function alignPool() {
  **/
 function Buffer(arg, encodingOrOffset, length) {
   // Common case.
+  if (pendingDeprecation && bufferWarn) {
+    // This is a *pending* deprecation warning. It is not emitted by
+    // default unless the --pending-deprecation command-line flag is
+    // used or the NODE_PENDING_DEPRECATION=1 envvar is set.
+    process.emitWarning(bufferWarning, 'DeprecationWarning', 'DEP0005');
+    bufferWarn = false;
+  }
+
   if (typeof arg === 'number') {
     if (typeof encodingOrOffset === 'string') {
       throw new Error(
         'If encoding is specified then the first argument must be a string'
       );
     }
-    return Buffer.allocUnsafe(arg);
+    assertSize(arg);
+    return new FastBuffer(createRandomFillBuffer(arg));
```

Comment:
@jasnell The best way that comes to mind for me right now is to have separate pools for uninitialized & for random-filled Buffers … alternatively, we could just keep using `Buffer.allocUnsafe()` here and fill manually.